src/base.ts

```diff
@@ -79,38 +75,68 @@ export const resolveServiceUrl = (service: Service): string => {
   throw new Error(`Missing endpoint configuration for "${service}" service`)
 }
 
-const authorizationHeader = () => {
-  if (!apiClient.tokens.accessToken) {
-    return null
+const getAuthorizationHeader = (accessToken = null) => {
+  if (!accessToken) {
+    throw new Error(
+      "You tried to make an authenticated requests without providing an access token!\n Please pass a valid token as a request option."
+    )
   }
 
-  return `Bearer ${apiClient.tokens.accessToken}`
+  return { Authorization: `Bearer ${accessToken}` }
+}
+
+export interface RequestOptions {
+  isAuthorizedRequest?: boolean
+  accessToken?: string
+  method?: "GET" | "POST" | "PUT" | "PATCH" | "DELETE"
+  headers?: HeadersInit
 }
 
-export const fetchPath = async (
-  service: Service,
-  path: string,
-  options = {}
-) => {
-  const prefix = resolveServiceUrl(service)
-  const url = `${prefix}/${stripLeadingSlash(path)}`
+export const fetchPath = async ({
+  service,
+  path,
+  body,
+  options = {},
+}: {
+  service: Service
+  path: string
```

Comment:
I would keep service/path as regular attributes - but really up to you. I think krzysztof is preferring named attributes as well

Reply:
The reason I went with this is there is an optional body attribute in the current interface on position 3 adding the options will result in a polymorpic interface where the options might be argument 3 or 4 have seen this in framework code where they try to verify that the last argument contains the options why I don't like a lot....
Further on it can be handy if you change the interface eg Krzysztof will probably drop the service so we don't need to shift argument; Same applies for extending it